src/styles/index.css

```diff
@@ -0,0 +1,393 @@
+  content: "22";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(23)::before {
+  content: "23";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(24)::before {
+  content: "24";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(25)::before {
+  content: "25";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(26)::before {
+  content: "26";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(27)::before {
+  content: "27";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(28)::before {
+  content: "28";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(29)::before {
+  content: "29";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(30)::before {
+  content: "30";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:nth-child(31)::before {
+  content: "31";
+  font-family: Arial, Helvetica, sans-serif;
+  font-size: 30px;
+  position: absolute;
+  top: 50%;
+  left: 50%;
+  transform: translate(-50%, -50%);
+}
+
+.calendar__day:hover {
+  cursor: pointer;
+  transform: translateY(-20px);
+  background-color: #ffbfcb;
+}
+
+.calendar--start-day-mon .calendar__day:first-child {
+  grid-column-start: 1;
+}
+
+.calendar--start-day-tue .calendar__day:first-child {
+  grid-column-start: 2;
+}
+
+.calendar--start-day-wed .calendar__day:first-child {
+  grid-column-start: 3;
+}
+
+.calendar--start-day-thu .calendar__day:first-child {
+  grid-column-start: 4;
+}
+
+.calendar--start-day-fri .calendar__day:first-child {
+  grid-column-start: 5;
+}
+
+.calendar--start-day-sat .calendar__day:first-child {
+  grid-column-start: 6;
+}
+
+.calendar--start-day-sun .calendar__day:first-child {
+  grid-column-start: 7;
+}
+
+.calendar__day .calendar--month-length-31 {
+  display: block;
+}
+
+.calendar--month-length-28 .calendar__day:nth-child(n+29) {
+  display: none;
```

Comment:
The usage of `nth-child(n+29)` is correct for hiding days that are not in the month. However, ensure that the rest of the calendar logic matches the requirements for the task. For example, check if the calendar correctly adjusts when starting on different weekdays and if all months are represented accurately.